### PR TITLE
feat: support basic CLI

### PR DESCRIPTION
**Pull request by bzp2010**
### What this PR does / why we need it:

This PR add basic CLI support to the project, and it will currently start an instance with MySQL as the backend by default (i.e. examples/mysql before). 

More features, such as configurability, will be provided in a later PR.

#### Other
ping @tokers PTAL, thanks.

### CODE REVIEW

File: test/shell/cli_test.sh
Comment (tokers) [inline test/shell/cli_test.sh:27]: Do not use such a mandatory way stop the process.

File: internal/backend/cache.go
Comment (tokers) [inline internal/backend/cache.go:16]: What's the meaning of this change?

File: cmd/root.go
Comment (tokers) [inline cmd/root.go:62]: Exportable methods/functions should have their comments to explain the usage.

File: cmd/root.go
Comment (tokers) [inline cmd/root.go:34]:
```suggestion
	Short: "The bridge between etcd protocol and other storage backends.",
```

File: cmd/root.go
Comment (tokers) [inline cmd/root.go:29]:
```suggestion
	"github.com/api7/etcd-adapter/internal/backend/mysql"
	"github.com/api7/etcd-adapter/internal/adapter"
```

File: internal/backend/cache.go
Comment (bzp2010) [reply internal/backend/cache.go]: changed back

File: cmd/root.go
Comment (bzp2010) [reply cmd/root.go:62]: added

File: cmd/root.go
Comment (bzp2010) [reply cmd/root.go]: changed

File: cmd/root.go
Comment (bzp2010) [reply cmd/root.go]: re-ordered

File: test/shell/cli_test.sh
Comment (bzp2010) [reply test/shell/cli_test.sh:27]: Hi, @tokers.

🤔 I'm sorry I'm a bit confused, the idea of this code comes from `apisix-dashboard`'s CLI test (`apisix-dashboard` uses systemd to manage process), in this project we haven't introduced systemd related code yet and it may not be supported in the future, so I'm currently using this way to force end process.

If there is a better way, I hope you can guide me. Thanks

Comment (tokers) [reply test/shell/cli_test.sh:27]: We can just handle the `SIGINT` and `SIGTERM` signals in our program, so that we can stop the program gracefully.

File: internal/backend/cache.go
Comment (tokers) [reply internal/backend/cache.go]: OK.

File: test/shell/cli_test.sh
Comment (bzp2010) [reply test/shell/cli_test.sh:27]: So, we use `SIGTERM` instead of `SIGKILL` to achieve a graceful exit of the program.

Comment (bzp2010) [reply test/shell/cli_test.sh]: changed, Now adapter will receive `SIGINT` and `SIGTERM` signal, print the log and exit. The test script has also been changed to send SIGTERM by default, but since the current logs are not output to a file, we can't check it for now. It will check later if the `See you next time!` are printed out.

ping @tokers

File: cmd/root.go
Comment (tokers) [inline cmd/root.go:60]: Should call adapter's `Shutdown` method.

Comment (bzp2010) [reply cmd/root.go:60]: fixed

**Review by tokers — APPROVED** [review]

**Comment by tokers** [comment]
@starsz please help us to take a look, thanks!

File: .github/workflows/cli.yaml
Comment (starsz) [inline .github/workflows/cli.yaml:11]: Missing timeout

File: .github/workflows/cli.yaml
Comment (starsz) [inline .github/workflows/cli.yaml:37]: For debug?

File: cmd/root.go
Comment (starsz) [inline cmd/root.go:45]: Why it's const value?

Comment (bzp2010) [reply cmd/root.go:45]: Yes, in this PR it is still Demo, in the next PR #6 it will support configurability.

File: .github/workflows/cli.yaml
Comment (bzp2010) [reply .github/workflows/cli.yaml:11]: I don't think this project needs to set a global timeout for CI, the open source project is not limited by the CI quota for the owner account (currently it is free) and it has a maximum timeout of 6 hours.

But I think it is necessary to add a timeout to the next tmate debugger, if it is not limited it will keep running until the CI timeout.

**Review by starsz — APPROVED** [review]

**Comment by bzp2010** [comment]
Hi, @tokers. Do we need to continue the review after we have already obtained the approve of another reviewer?

**Pull request merged**
